src/org/pahappa/systems/services/IncidentServiceImpl.java

```diff
@@ -86,9 +86,22 @@ public Incident getIncidentOfId(int id) {
 
 	@Override
 	public void deleteIncident(Incident incident) {
-		// TODO Auto-generated method stub
-		
+		String incidentToDelete = incident.getTitle();
+		int incidentSize = incidents.size();
+
+		for (Incident item : incidents) {
+			if(incidentToDelete == item.getTitle()){
+				incidents.remove(item);
+				System.out.println("The incident has been successfully deleted");
+				break;
+			}
+		}
```

Comment:
You can solve this in a better way than iterating through the list to have a more readable code.

- Check if the incident given is not null
- Then make use `.remove` function on that list to delete that incident

Reply:
Ohhh, yes. that is better. Thank you.

Reply:
I've update the function